org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -79,149 +91,272 @@
+							}
 
-						if (needInterp)
-						{
-							// we now need to decide which set of indices to use as the master
-							final boolean useAforIndexMaster;
+							// ok, they're both monotonic, in the same direction. we can continue.
+						}
 
-							// find the data limits
-							final double aMin = aIndices.min().doubleValue();
-							final double aMax = aIndices.max().doubleValue();
+						// find the data limits
+						final double aMin = aIndices.min().doubleValue();
+						final double aMax = aIndices.max().doubleValue();
+						final double bMin = bIndices.min().doubleValue();
+						final double bMax = bIndices.max().doubleValue();
 
-							final double bMin = bIndices.min().doubleValue();
-							final double bMax = bIndices.max().doubleValue();
+						if (aMax < bMin || aMin > bMax)
+						{
+							// datasets don't overlap
+							throw new IllegalArgumentException(INDICES_DO_NOT_OVERLAP);
+						}
+						else if (aMin <= bMin && aMax >= bMax)
+						{
+							// ok, b is wholly contained within A. Just generate
+							// points in A at the time in B
+							final Dataset interpolatedValues =
+									Maths.interpolate(aIndices, da, bIndices, null, null);
+							// remember the output axes, since we'll put them
+							// into the results
+							interpolatedValues.addMetadata(axesB);
+
+							operandA = db;
+							operandB = interpolatedValues;
+						}
+						else if (bMin < aMin && bMax > aMax)
+						{
+							// ok, A is wholly contained within B. Just generate
+							// points in B at the time in A
+							final Dataset interpolatedValues =
+									Maths.interpolate(bIndices, db, aIndices, null, null);
+							// remember the output axes, since we'll put them
+							// into the results
+							interpolatedValues.addMetadata(axesA);
 
-							if (aMax < bMin || aMin > bMax)
-							{
-								// datasets don't overlap
-								throw new IllegalArgumentException(
-										"The indices of the dataset do not overlap");
-							}
-							else if (aMin <= bMin && aMax >= bMax)
+							operandA = da;
+							operandB = interpolatedValues;
+						}
+						else
+						{
+							// ok, one isn't contained within the other,
+							// they merely overlap. Find the overlapping period
+							final double startPoint = aMin < bMin ? bMin : aMin;
+							final double endPoint = aMax > bMax ? bMax : aMax;
+
+							// to find the last matching entry, we have to work through
+							// the dataset in reverse.
+							final Dataset aReverse = aIndices.getSlice(null, null, new int[]
+							{-1});
+							final Dataset bReverse = bIndices.getSlice(null, null, new int[]
+							{-1});
+
+							// now get the two slices in this period
+							final int aStart;
+							final int aEnd;
+							final int bStart;
+							final int bEnd;
+							if (aMono == Monotonicity.STRICTLY_INCREASING)
 							{
-								// ok, use b, since it fits within a
-								useAforIndexMaster = false;
+								// ok - data ascending
```

Comment:
In this strictly increasing case, you don't need the reversed indices as you are guaranteed that each value exists only once